lib/temporal/client.rb

```diff
@@ -229,25 +229,29 @@ def await_workflow_result(workflow, workflow_id:, run_id: nil, timeout: nil, nam
       execution_options = ExecutionOptions.new(workflow, options, config.default_execution_options)
       max_timeout = Temporal::Connection::GRPC::SERVER_MAX_GET_WORKFLOW_EXECUTION_HISTORY_POLL
       history_response = nil
-      begin
-        history_response = connection.get_workflow_execution_history(
-          namespace: execution_options.namespace,
-          workflow_id: workflow_id,
-          run_id: run_id,
-          wait_for_new_event: true,
-          event_type: :close,
-          timeout: timeout || max_timeout,
-        )
-      rescue GRPC::DeadlineExceeded => e
-        message = if timeout
-          "Timed out after your specified limit of timeout: #{timeout} seconds"
-        else
-          "Timed out after #{max_timeout} seconds, which is the maximum supported amount."
+      while(true) do
```

Comment:
Just a suggestion to use some sugar:

```suggestion
      loop do
```

Reply:
Done!